tests/tools/cadd/cadd.py

```diff
@@ -0,0 +1,115 @@
+""" Module provides interface to web APIs of CADD tool. """
+import argparse
+
+import pandas as pd
+import requests
+
+
+class BadResponseException(Exception):
+    """Custom exception for bad responses."""
+
+
+class DownloadError(Exception):
+    """Custom exception for download errors."""
+
+
+def fetch_cadd_scores(cadd_version, chromosome, start, end=None):
+    """
+    Fetches CADD (Combined Annotation Dependent Depletion) scores for either a single SNV or a range of genomic
+    positions.
+
+    :param str cadd_version: Version of the CADD model used, e.g., "v1.3" or "GRCh38-v1.7".
+    :param int chromosome: Chromosome number where the SNV or genomic region is located.
+    :param int start: Genomic start position (or single position for SNV) of the region.
+    :param int end: (Optional) Genomic end position of the region. If not provided, fetches a single SNV.
+    :return: A dictionary containing CADD scores and annotations for the specified SNV or region, or None if an
+    error occurs.
+    """
+
+    if end:
+        url = f"https://cadd.gs.washington.edu/api/v1.0/{cadd_version}/{chromosome}:{start}-{end}"
+    else:
+        url = f"https://cadd.gs.washington.edu/api/v1.0/{cadd_version}/{chromosome}:{start}"
+
+    try:
+        response = requests.get(url, timeout=30)
+        if response.status_code == 200:
+            data = response.json()
+            return data
+        raise BadResponseException(f"Error: {response.status_code} - {response.text}")
+    except requests.exceptions.Timeout as exc:
+        raise DownloadError("Error: Timeout occurred while trying to reach the server.") from exc
+    except requests.exceptions.RequestException as req_err:
+        raise DownloadError(f"Error: {req_err}") from req_err
+    except ValueError as exc:
+        raise BadResponseException("Error: Invalid JSON format in response.") from exc
```

Comment:
Purpose of try/catch:
1. Simplify debugging
2. Make sure resources are deallocated

In line 42-43 you don't explain reason of error, you just copy message.